### PR TITLE
feat: get VehicleLockStatus

### PR DESCRIPTION
**Pull request by naomiperez**
This PR will add support for getting the lock status of a vehicle.

We create a new method in the `Vehicle` class called `lockStatus()` that will return a `VehicleLockStatus` object containing a property `isLocked` (true if vehicle is locked), and arrays of
- `VehicleDoor`
- `VehicleWindow`
- `VehicleStorage`
- `VehicleSunroof`
- and `VehicleChargingPort`

each of which have a `type` and `status`.

Asana: https://app.asana.com/0/1205281929337658/1205139255747896/f

### CODE REVIEW

File: src/main/java/com/smartcar/sdk/Vehicle.java [hunk 1]
```diff
@@ -359,6 +359,16 @@ public ActionResponse stopCharge() throws SmartcarException {
     return this.call("charge", "POST", body, ActionResponse.class);
   }
 
+  /**
+   * Send request to the /security endpoint to get the vehicle's lock status
+   *
```
Comment (wikiadrian) [inline src/main/java/com/smartcar/sdk/Vehicle.java:364]: Returns VehicleLockStatus

**Review by wikiadrian — COMMENTED** [review]
The PR LGTM The only question I have is if we are going  to add the batch support in this PR or in a different one

**Review by wikiadrian — APPROVED** [review]

File: src/main/java/com/smartcar/sdk/data/VehicleChargingPort.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package com.smartcar.sdk.data;
+
+public class VehicleChargingPort {
```
Comment (JacobAndrewSmith92) [inline src/main/java/com/smartcar/sdk/data/VehicleChargingPort.java:3]: question: does this class not need to extend `ApiData`?

**Review by naomiperez — COMMENTED** [review]
Looks good!

**Review by JacobAndrewSmith92 — COMMENTED** [review]
LGTM 

General question: Does it make sense to create a common class that these values can inherit from?

All of these classes have a a `type and status String`.

I don't know if it's worth it or not as I'm not the Java expert.

File: src/main/java/com/smartcar/sdk/data/VehicleChargingPort.java (hunk 2 above)
Comment (wikiadrian) [reply src/main/java/com/smartcar/sdk/data/VehicleChargingPort.java:3]: Good catch!!. It needs the ApiData

**Comment by wikiadrian** [comment]
The question about having a parent class with type and status. 
I think this depends of the level of abstraction we are doing. When I specced the tickets I tried to come up with a generic class that wraps all the concepts but none of the components share any other attribute or a term in common. The only shared the openability status and all belong to a car, the former  is more a composition relation.

**Comment by JacobAndrewSmith92** [comment]
> The question about having a parent class with type and status. I think this depends of the level of abstraction we are doing. When I specced the tickets I tried to come up with a generic class that wraps all the concepts but none of the components share any other attribute or a term in common. The only shared the openability status and all belong to a car, the former is more a composition relation.

Makes sense. Yeah, I was alluding to a composition type implementation when I was thinking about it. I think it doesn't really matter here. Just wanted to discuss. I like your reasoning.

**Pull request merged**